gstd/src/async_runtime/signals.rs

```diff
@@ -90,7 +90,11 @@ impl WakeSignals {
             Some(WakeSignal {
                 payload: Some(payload),
                 ..
-            }) => ReplyPoll::Some(payload),
+            }) => {
+                // Remove lock after waking.
+                super::locks().remove(&crate::msg::id());
```

Comment:
Seems that's not correct place for this call (as minimum, usage should be somewhere in place `wait/wake/poll` called, further - it produces same bug as I mentioned before)